tests/inventory/pipelines/base_pipeline_test.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests the base pipeline."""
+
+
+from google.apputils import basetest
+import mock
+
+# pylint: disable=line-too-long
+from google.cloud.security.common.data_access import dao
+from google.cloud.security.common.data_access import errors as data_access_errors
+from google.cloud.security.common.gcp_api import cloud_resource_manager as crm
+from google.cloud.security.common.gcp_api import errors as api_errors
+from google.cloud.security.common.util import parser
+from google.cloud.security.common.util.log_util import LogUtil
+from google.cloud.security.inventory import errors as inventory_errors
+from google.cloud.security.inventory.pipelines import load_projects_pipeline
+from tests.inventory.pipelines.test_data.fake_projects import EXPECTED_LOADABLE_PROJECTS
+# pylint: enable=line-too-long
+
+
+class BasePipelineTest(basetest.TestCase):
+    """Tests for the base pipeline.
+    
+    Because base pipeline is an abstract class, there is no way to instantiate
+    it for testing directly.  So, will test it by a pipeline that implements it.
+    """
+
+    def setUp(self):
+        """Set up."""
+
+        self.cycle_timestamp = '20001225T120000Z'
+        self.configs = {'organization_id': '66666',
```

Comment:
You could choose to move the config dict into your test_data as well instead of copying it into each test class.